INSTALL.md

```diff
@@ -0,0 +1,86 @@
+
+FACT consists of three components: frontend, database and backend. All components can be installed on one machine as well as on different machines.
+There is an automated installation program supporting Ubuntu 16.04 and 18.04 host systems.
+
+## TL;DR
+
+0. `(cd path/to/fact/sources && ./pre_install.sh)`
+1. `sudo reboot`
+2. `(sudo mkdir /media/data && sudo chown -R $USER /media/data)`
+3. `(cd path/to/fact/sources && ./install.py)`
+
+Start with `./start_all_installed_fact_components`.
+
+## Simple One System Setup
+
+### Pre-Install
+
+Some components of FACT depend on docker. Since a restart is necessary before docker can run. The system has to be rebooted in between FACT installation. Thus the docker setup and some auxiliary stuff has been moved to the `pre_install.sh` script.
+
+The following lines will handle the first half of the installation (based on git not tarball):
+
+```sh
+$ sudo apt-get install git
+$ git clone https://github.com/fkie-cad/FACT_core.git
+$ cd FACT_core
+$ src/bootstrap/pre_install.sh
+```
+
+Now modify *src/config/main.cfg* and *src/config/mongod.conf* to suit your needs.
+Especially, you should change the mongo passwords.
+The database is initialized with these passwords on first start.  
+
+Create the mongo data directory specified in *mongod.conf* and the firmware_file_storage_directory defined in *main.cfg*.
+Make sure that the log directory exists as well.
+
+If you have any additional plug-ins, copy/clone them into corresponding *src/plugins/* directory.  
+
+:exclamation: **Reboot before executing the ./install.py** :exclamation:  
+:exclamation: **You have to do the above steps before you do anything else** :exclamation:
+
+### Main-Install
+
+:customs: **The installation script installs a lot of dependencies that may have different licenses**
+  
+After reboot, enter the FACT_core directory and execute:  
+
+```sh
+$ ./install.py
+```
+
+:beer: **Get a drink... Installation of the dependencies might take some time...** :tea:
+
+For more advanced setups have a look at the help function of the installer:
+
+```sh
+$ ./install.py --help
+```
+
+## Multi system setup
+
+The two worker components (frontend, backend) communicate exclusively through the database. The database in turn does not needed any knowledge of its place in the network, other than on which **ip:port** combination the database server has to be hosted.
+The main.cfg on the frontend system has to be altered so that the values of `data_storage.mongo_server` and `data_storage.mongo_port` match the **ip:port** for the database.
+The same has to be done for the backend. In addition, since the raw firmware and file binaries are stored in the backend, the `data_storage.firmware_file_storage_directory` has to be created (by default `/media/data/fact_fw_data`).
+On the database system, the `mongod.conf` has to be given the correct `net.bindIp` and `net.port`. In addition the path in `storage.dbPath` of the `mongod.conf` has to be created. 
+
```

Comment:
The install options should be explained as well ./install.py [-B -D -F]
Furhtermore we should mention that the databse should be started first.

Reply:
I disagree with the second request. Starting the database first - or at all - has nothing to do with installation but execution. This should be documented in a getting started page.